projects/commudle-admin/src/app/app.component.ts

```diff
@@ -62,7 +66,17 @@ export class AppComponent {
           //   this.sidebarService.expand('mainMenu');
           // }
         }
-      });
+      }); 
+  }
+
+  ngOnInit() {
+    if (!this.cookieConsentService.isCookieConsentAccepted()) {
+      this.openWindow();
+    }
+  }
+
+  openWindow() {
+    this.windowService.open(CookieConsentComponent, { title: 'Cookie Consent', hasBackdrop: false, initialState: NbWindowState.MAXIMIZED});
   }
```

Comment:
I think we should move these lines to the cookie consent component itself and call the component in the app component html. The window will then be opened using ng-template [[Window Content from TemplateRef](https://akveo.github.io/nebular/docs/components/window/overview#nbwindowservice)]

Reply:
Hello! I'm happy to make this change however I don't think I quite understand what you're suggesting here.  

How would you call the component in the app.component.html? The example for the Nebular documentation that you linked to still has the logic to openWindow in the same component that is opening the ng-template.

I'm struggling with what triggers opening the window if we move the openWindow functionality to the cookie consent component away from app.component?

Also, why would we want to check if cookie consent has been accepted inside the cookie consent component?  Wouldn't we want to check the consent BEFORE we open the window during initial loading of the app?